corehq/apps/registration/utils.py

```diff
@@ -96,25 +97,32 @@ def safe_unsubscribe_user_from_mailchimp_list(user, list_id, email=None):
 
 def handle_changed_mailchimp_email(user, old_email, new_email):
     """
-        Checks whether there are other users with old_email who are also subscribed to any mailchimp lists.
-        If not, it safely unsubscribes that email from mailchimp. Then, adds new_email to mailchimp.
+    Checks whether there are other users with old_email who are also subscribed to any mailchimp lists.
+    If not, it safely unsubscribes that email from mailchimp. Then, adds new_email to mailchimp.
     """
-    users_with_old_email = User.objects.filter(email=old_email).values_list('username', flat=True)
 
-    couch_users = CouchUser.view('users/by_username',
-                                 keys=list(users_with_old_email),
-                                 include_docs=True,
-                                 reduce=False,
-                                 ).all()
+    @memoized
```

Comment:
yup, `@memoized` just literally adds a dict of input vars to return value to this function (think, `get_users_who_have_email._cache = {}`), so when the function gets garbage collected so will that (unless it's kept alive by something else pointing to it, which is unlikely/rare). In the case of an instance method, the cache goes on `self` instead, to tie the lifecycle to the instance rather than the class—but that's a special case.